### PR TITLE
feat(v2): bumping typescript target and library to es2022

### PR DESCRIPTION
**Pull request by eranhirsch**
Just doing this change and following up on any errors and warnings

### CODE REVIEW

File: tsconfig.dist.json [hunk 1]
```diff
@@ -8,9 +8,6 @@
 
     "noEmit": true,
 
-    "target": "ES5",
-    "lib": ["ES6", "ES2017.Object", "DOM", "DOM.Iterable"],
-
```
Comment (cjquines) [inline tsconfig.dist.json:13]: **question:** why remove this? my understanding is that we use this to typecheck the built output in ci, and we need the lib definitions to do so

Comment (eranhirsch) [reply tsconfig.dist.json:13]: Looking at the CI there are issues with this, I might revert it. It seemed redundant when I looked at it but I might have just gotten something wrong... I thought this file inherits/extends from the main one and these are just duplications.

Comment (cjquines) [reply tsconfig.dist.json:13]: oh, no, it's an entirely separate ts file...

**Pull request merged**